### PR TITLE
Replacing unreliable array initialization in SiPixelTemplateReco.cc

### PR DESCRIPTION
**Pull request by uhussain**
arrays nyzero and ysum initialized to zero without using the for loop in line 220. The for loop is unreliable because the bound of the loop is an input argument that need not match the size of the array

### CODE REVIEW

File: RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc [hunk 1]
```diff
@@ -216,8 +216,8 @@ int SiPixelTemplateReco::PixelTempReco2D(int id, float cotalpha, float cotbeta,
 	
     if(deadpix) {
 	   fypix = BYM3; lypix = -1;
-       for(i=0; i<nclusy; ++i) {
-	      ysum[i] = 0.f; nyzero[i] = 0;
+       	   memset(nyzero, 0, TYSIZE * sizeof(int));
+           memset(ysum, 0, BYSIZE * sizeof(float));
 // Do preliminary cluster projection in y
 	      for(j=0; j<nclusx; ++j) {
 		     ysum[i] += cluster(j,i);
```
Comment (VinInn) [inline RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc]: sorry, where "i" is now supposed to come in this loop?

**Comment by VinInn** [comment]
could you be more specific about "unreliable"?

File: RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc (hunk 1 above)
Comment (uhussain) [inline RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc]: yes the for loop should not have been removed, thanks for pointing it out.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

For #15386 c8fbecdc2f5bea5052076b093d26cd9590398606

Small fix in SiPixelTemplateReco.cc for a static analyzer warning to ensure arrays are properly initialized in all cases. There should be no change in monitored quantities.

The code fix is satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-08-08-1100 show no significant differences, as expected.

**Comment by cvuosalo** [comment]
@uhussain: Could you please revise the title of this PR? It should be: "Replacing unreliable array initialization in SiPixelTemplateReco.cc". A more accurate title might help get this PR merged faster.

**Comment by davidlange6** [comment]
+1
the indentation of this routine is a mess (but this PR doesn't change that)

**Pull request merged**